CodeChat_Server/windows_installer.ps1

```diff
@@ -0,0 +1,205 @@
+# |
+# 
+# Location of the Version of Python we are downloading (3.9.12)
+#   We're using 3.9 because as far as I remember there were some weird bugs with 3.10
+#   ``$PyTarget`` is intended to be changed if a new version target is identified i.e 3.10 or newer is not buggy
+$PyTarget = "3.9.12"
+# Location of Python install exe
+$PythonURL = "https://www.python.org/ftp/python/3.9.12/python-$PyTarget-amd64.exe"
+# |
+# 
+# Where we are downloading the installer, we put the name of the file because Invoke-WebRequest needs this to output to the location.
+$PythonDest = "$env:USERPROFILE\Downloads\python-$PyTarget-amd64.exe"
+# |
+# 
+# Seeing if any Version of Python is Installed
+# --------------------------------------------
+# Get-Command -all will search for any python executable (denoting an installation)
+$PythonVersion = Get-Command -all python -erroraction 'silentlycontinue'
+# If the command does not return a string (with the python.exe) we have to do some stuff
+if([string]::IsNullOrEmpty($pythonVersion)){
+    # Asking the user if they would like to have python installed on their system
+    $response = Read-Host -Prompt "You do not have Python installed, would you like to install the required version ($pythonVersionReq) (y/n)?"
+    if($response -eq "y"){
+        echo "Downloading Python Version $PyTarget"
+        # Downloading Python Unsure if ``-UseDefaulCredential`` is necessary
+        # Using Invoke-WebRequest because I couldn't get the other method outlined in the `tutorial <https://lazyadmin.nl/powershell/download-file-powershell>`_ working guide says it's faster though...
+        Invoke-WebRequest -Uri $PythonURL -OutFile $PythonDest -UseDefaultCredential
+        echo "Python Version $PyTarget Downloaded
+        Installing Python"
+        Read-Host "Press any button to continue, then click 'Install Now' in popup"
+        # 
+        # Redirecting to downloads so that we can run our installation
+        cd $env:USERPROFILE\Downloads
+        # 
+        # Installs Python ``Out-Null`` option makes sure the program doesn't continue before the installation completes, The ``/quietly`` just makes no window pop up, which is fine because ``PrependPath=1`` will select the PATH checkbox for us.  
+        # Potentially Install-Package could be a better fit, not sure if it only works on Powershell Modules or other exe's too
+        Start-Process ".\python-$PyTarget-amd64.exe" -ArgumentList /quietly,PrependPath=1 -Wait
+        echo "Python Version $PyTarget Installed Successfully
+        "
+    }
+    # If they don't have python and they don't want us to do it all we can do is let them know what the minimum version to get is
+    else{
+        echo "Please install the minimum required version of python ($pythonVersionReq) or later
+        (Be sure to add to the PATH)"
+        exit
+    }
+}
+# Returning to User Dir
+cd $env:USERPROFILE
+# |
+# 
+# Grabbing the version of python and splitting it up so as to make parsing easier
```

Comment:
Use the nice Python script from `codechat.sh` here (it's a bit shorter and simpler than this.)